### PR TITLE
convert dt.c to D

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/dmd/backend/dt.d [hunk 1]
```diff
@@ -5,75 +5,616 @@
...
+
+void dt2common(dt_t **pdt)
+{
+    assert((*pdt).dt == DT_azeros);
+    (*pdt).dt = DT_common;
+}
+
+/**********************************************************/
+
+struct DtBuilder
```
Comment (jacob-carlborg) [inline src/dmd/backend/dt.d]: Why was this changed to a struct?

Comment (thewilsonator) [reply src/dmd/backend/dt.d]: presumably because it is no longer used from C++. Why not a final class, I'm not sure.

Comment (jacob-carlborg) [reply src/dmd/backend/dt.d]: It goes against Walter's rules about translating code, that's why I was asking.

Comment (thewilsonator) [reply src/dmd/backend/dt.d]: Good question. Dunno.

Comment (WalterBright) [reply src/dmd/backend/dt.d]: Classes require TypeInfo, which is not allowed in -betterC. This needs to be better, but for the time being we are stuck with the bootstrap compiler as it is.

Comment (WalterBright) [reply src/dmd/backend/dt.d]: Also, .di files do not work with -betterC, even with structs, so I had to abandon that as well.

Comment (thewilsonator) [reply src/dmd/backend/dt.d]: Fair enough.

**Review by thewilsonator — APPROVED** [review]
